src/utils/BlockUtils.ts

```diff
@@ -101,7 +106,7 @@ export class BlockFinder {
    * @notice Gets the latest block whose timestamp is <= the provided timestamp.
    * @param {number} timestamp timestamp to search.
    */
-  public async getBlockForTimestamp(timestamp: number | string): Promise<Block> {
+  public async getBlockForTimestamp(timestamp: number | string, hints: BlockFinderHint): Promise<Block> {
```

Comment:
tiny nit: I think hint is singular.

```suggestion
  public async getBlockForTimestamp(timestamp: number | string, hint: BlockFinderHint): Promise<Block> {
```

Reply:
I tried this but reverted to using plural everywhere, so at least it's consistent.